utils/s3/creds.hh

```diff
@@ -16,6 +16,7 @@ namespace s3 {
 struct endpoint_config {
     unsigned port;
     bool use_https;
+    bool is_default;
```

Comment:
Adding new configuration property should come in its own patch (e.g. as a part of this PR, but can be its own PR as well)